sqlx-sqlite/src/conversions.rs

```diff
@@ -0,0 +1,229 @@
+            ScalarValue::Float32(v) => SimpleExpr::Constant(Value::Float(*v)),
+            ScalarValue::Float64(v) => SimpleExpr::Constant(Value::Double(*v)),
+            ScalarValue::Int8(v) => SimpleExpr::Constant(Value::TinyInt(*v)),
+            ScalarValue::Int16(v) => SimpleExpr::Constant(Value::SmallInt(*v)),
+            ScalarValue::Int32(v) => SimpleExpr::Constant(Value::Int(*v)),
+            ScalarValue::Int64(v) => SimpleExpr::Constant(Value::BigInt(*v)),
+            ScalarValue::UInt8(v) => SimpleExpr::Constant(Value::TinyUnsigned(*v)),
+            ScalarValue::UInt16(v) => SimpleExpr::Constant(Value::SmallUnsigned(*v)),
+            ScalarValue::UInt32(v) => SimpleExpr::Constant(Value::Unsigned(*v)),
+            ScalarValue::UInt64(v) => SimpleExpr::Constant(Value::BigUnsigned(*v)),
+            ScalarValue::Utf8(v) => match v {
+                Some(v) => SimpleExpr::Constant(Value::String(Some(Box::new(v.to_string())))),
+                None => SimpleExpr::Constant(Value::String(None)),
+            },
+            ScalarValue::LargeUtf8(v) => match v {
+                Some(v) => SimpleExpr::Constant(Value::String(Some(Box::new(v.to_string())))),
+                None => SimpleExpr::Constant(Value::String(None)),
+            },
+            ScalarValue::Binary(v) => match v {
+                Some(v) => SimpleExpr::Constant(Value::Bytes(Some(Box::new(v.to_vec())))),
+                None => SimpleExpr::Constant(Value::Bytes(None)),
+            },
+            ScalarValue::FixedSizeBinary(_, v) => match v {
+                Some(v) => SimpleExpr::Constant(Value::Bytes(Some(Box::new(v.to_vec())))),
+                None => SimpleExpr::Constant(Value::Bytes(None)),
+            },
+            ScalarValue::LargeBinary(v) => match v {
+                Some(v) => SimpleExpr::Constant(Value::Bytes(Some(Box::new(v.to_vec())))),
+                None => SimpleExpr::Constant(Value::Bytes(None)),
+            },
+            // Extend with other types, e.g. to support arrays, dates, etc.
+            _ => SimpleExpr::Constant(Value::Bool(Some(true)))
+        }
+    } else if let Some(expr) = expr.as_any().downcast_ref::<phys_expr::CaseExpr>() {
+        let mut case = CaseStatement::new();
+        for (when, then) in expr.when_then_expr() {
+            case = case.case(physical_expr_to_sea_query(when), physical_expr_to_sea_query(then));
+        }
+        if let Some(else_exp) = expr.else_expr() {
+            case = case.finally(physical_expr_to_sea_query(else_exp));
+        };
+        SimpleExpr::Case(Box::new(case))
+    } else {
+        SimpleExpr::Constant(Value::Bool(Some(true)))
+    }
+}
+
+/// Convert a DataFusion Array to a Vec of SeaQuery Values
+pub fn array_to_values(array: ArrayRef) -> Option<Vec<Value>> {
+    let values = match array.data_type() {
+        DataType::Int8 => {
+            let array = array.as_primitive::<datatypes::Int8Type>();
+            array.iter().map(|v| {
```

Comment:
Can the map closure here be simplified to:

`array.iter().map(Value::TinyInt).collect()`

and similar for the other data types?

Reply:
Yes, at least those that don’t require a box